src/main/java/net/dries007/tfc/common/blocks/wood/ToolRackBlock.java

```diff
@@ -133,9 +141,35 @@ protected void createBlockStateDefinition(StateDefinition.Builder<Block, BlockSt
         builder.add(FACING, WATERLOGGED);
     }
 
+    @Override
+    @SuppressWarnings("deprecation")
+    public InteractionResult use(BlockState state, Level level, BlockPos pos, Player player, InteractionHand hand, BlockHitResult hit)
+    {
+        ToolRackBlockEntity te = Helpers.getBlockEntity(level, pos, ToolRackBlockEntity.class);
```

Comment:
Helpers.getBE is deprecated (you had to suppress the deprecation on use so you didn't notice)

use Level#getBlockEntity(pos, type) and resolve the optional